### PR TITLE
[FIX][12.0] _add_missing_default_values fixed to work with _log_access=False

### PR DESCRIPTION
**Pull request by katyukha**
#### Description of the issue/feature this PR addresses:
Fix `_add_missing_default_values` to make it work if model has `_log_access=False`

There is statement that creates set with field names that have to have default vallues in this method:
```python
missing_defaults = {
    name
    for name, field in self._fields.items()
    if name not in values
    if self._log_access and name not in MAGIC_COLUMNS   # This line is buggy
    if not (field.inherited and field.related_field.model_name in avoid_models)
}
```

Take a look at line `if self._log_access and name not in MAGIC_COLUMNS`, that means: if we have `_log_access` set to `False` then we will skip all fields, thus no default values will be computed.

Also, take attention, that [in odoo 13 this line is removed completely](https://github.com/odoo/odoo/blob/13.0/odoo/models.py#L1719).

#### Current behavior before PR:
Define model with disabled log access:

```python
class MyModel(models.Model):
    _name = 'my.model'
    _log_access = False

    name = fields.Char()
    test_field = fields.Char(default='Test Default')
```

Try to create record, assuming that field `test_field` will be automatically set to default value 'Test Default', and test if default value was computed.

```python
rec = self.env['my.model'].create({'name': 'test record'})
self.assertEqual(rec.test_field, 'Test Default')
```

Before this PR, test will fail, because `rec.test_field` will not be set to default value.

#### Desired behavior after PR is merged:
Define model with disabled log access:

```python
class MyModel(models.Model):
    _name = 'my.model'
    _log_access = False

    name = fields.Char()
    test_field = fields.Char(default='Test Default')
```

Try to create record, assuming that field `test_field` will be automatically set to default value 'Test Default', and test if default value was computed.

```python
rec = self.env['my.model'].create({'name': 'test record'})
self.assertEqual(rec.test_field, 'Test Default')
```
Test passes

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by mart-e** [comment]
@rco-odoo this line was removed at 9920f20e4c7753bc17bea71dea3a90f7de687196
Could you confirm we can change this in 12.0? It looks good to me.

@katyukha could you please check your diff? You remove some trailing spaces, it makes the diff more difficult to read, just keep it to the minimum please.

**Comment by katyukha** [comment]
Hi,

Thanks for reviewing this Pull Request.

I have rebased this PR and removed all changes related to trailing spaces.

**Comment by Feyensv** [comment]
@fw-bot up to saas-12.3

**Comment by katyukha** [comment]
Hi @Feyensv,
What is the current state of this PR? Is it planned to be merged into 12.0?

I am asking, because i have to decide, whether to wait while this PR is merged, or thinking about workaround for it in module level.

Thanks

File: odoo/models.py [hunk 1]
```diff
@@ -1680,7 +1680,7 @@ def _add_missing_default_values(self, values):
             name
             for name, field in self._fields.items()
             if name not in values
-            if self._log_access and name not in MAGIC_COLUMNS
+            if not self._log_access or (self._log_access and name not in MAGIC_COLUMNS)
```
Comment (rco-odoo) [inline odoo/models.py]: Could be simpler:
```suggestion
            if not self._log_access or name not in MAGIC_COLUMNS
```

**Comment by Feyensv** [comment]
I couldn't update the current PR, but this will be merged in 11.0 (and the following versions) through #45051 ;).

**Comment by katyukha** [comment]
Ok, thank you)

**Comment by rco-odoo** [comment]
See #45051.

**Pull request closed without merging**